lib/jwt.service.ts

```diff
@@ -9,19 +9,30 @@ export class JwtService {
     @Inject(JWT_MODULE_OPTIONS) private readonly options: JwtModuleOptions
   ) {}
 
-  sign(payload: string | Object | Buffer): string {
-    return jwt.sign(
-      payload,
-      this.options.secretOrPrivateKey,
-      this.options.signOptions
-    );
+  sign(payload: string | Object | Buffer, options?: jwt.SignOptions): string {
+    const signOptions = options
+      ? {
+          ...this.options.signOptions,
```

Comment:
note that `signOptions` is optional (it might be `undefined`)
use `...(this.options.signOptions || {})`

Reply:
You're totally right. I'll update the PR in a few minutes.